kamu-adapter-http/tests/tests/tests_pull/test_smart_pull_shared.rs

```diff
@@ -139,12 +139,18 @@ pub async fn test_smart_pull_existing_evolved_dataset<T: ServerSideHarness>(serv
     let client_harness = ClientSideHarness::new();
     let client_dataset_layout = client_harness.dataset_layout("foo");
 
-    // Hard folder synchronization
-    copy_folder_recursively(
```

Comment:
"Hard folder synchronization" is better to avoid as different repos have different "guarantees" on how they store data.

LocalFS repo always pre-creates all directories (`/blocks`, `/refs`, `/info`...) and expects them to exist. While in S3 there's no such thing as directory, so copying from later to former - we are breaking the expectation.

Rather than manipulating files I replaced them with syncs which should be more robust. Ideally I would avoid any direct file manipulations in these tests in the long run, and use `ObjectRepository` interfaces for copying data.

Reply:
You are right - let's adopt the policy of "objects, not files and directories" throughout all repositories.

Reverted these changes.